app/models/alchemy/page/page_elements.rb

```diff
@@ -28,23 +44,61 @@ module ClassMethods
       #
       def copy_elements(source, target)
         new_elements = []
-        source.elements.not_trashed.each do |element|
-          # detect cell for element
-          if element.cell
-            cell = target.cells.detect { |c| c.name == element.cell.name }
-          else
-            cell = nil
+        source.elements.not_trashed.each do |source_element|
+          cell = nil
+          if source_element.cell
+            cell = target.cells.find_by(name: source_element.cell.name)
           end
-          # if cell is nil also pass nil to element.cell_id
-          new_element = Element.copy(element, :page_id => target.id, :cell_id => (cell.blank? ? nil : cell.id))
-          # move element to bottom of the list
+          new_element = Element.copy source_element, {
+            page_id: target.id,
+            cell_id: cell.try(:id)
+          }
           new_element.move_to_bottom
           new_elements << new_element
         end
         new_elements
       end
+    end
 
+    # Finds elements of page.
+    #
+    # @param [Hash]
+    #   options hash
+    # @param [Boolean] (false)
+    #   Pass true, if you want to also have not published elements.
+    #
+    # @option options [Array] only
+    #   Returns only elements with given names
+    # @option options [Array] except
+    #   Returns all elements except the ones with given names
+    # @option options [Fixnum] count
+    #   Limit the count of returned elements
+    # @option options [Fixnum] offset
+    #   Starts with an offset while returning elements
+    # @option options [Boolean] random (false)
+    #   Return elements randomly shuffled
+    # @option options [Alchemy::Cell || String] from_cell
+    #   Return elements from given cell
+    #
+    # @return [ActiveRecord::Relation]
+    #
+    def find_elements(options = {}, show_non_public = false)
```

Comment:
Metrics/AbcSize: Assignment Branch Condition size for find_elements is too high. [21.73/15]